indexing-service/src/main/java/org/apache/druid/indexing/common/actions/RetrieveUsedSegmentsAction.java

```diff
@@ -85,6 +103,8 @@ public RetrieveUsedSegmentsAction(
 
     // Defaulting to the former behaviour when visibility wasn't explicitly specified for backward compatibility
     this.visibility = visibility != null ? visibility : Segments.ONLY_VISIBLE;
+
+    this.replace = replace != null ? replace : false;
```

Comment:
Can this not be determined automatically while running the action? Only tasks with a `REPLACE` lock would have this as true, right?

Reply:
Thanks, I think this can always be true.
If we need to fetch all the unfiltered segments for a task holding replace locks, we could use a task action client created using a different task that holds no locks.